src/calls/components/CallsView/Calendar.jsx

```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react';
+import {
+    startOfMonth,
+    getDaysInMonth,
+    getDay,
+    getYear,
+    format,
+    addMonths,
+    subMonths,
+    addDays,
+    setMonth,
+    setYear
+} from 'date-fns';
+
+class Calendar extends Component {
+    state = {
+        date: new Date(),
+        months: [
+            'Jan',
+            'Feb',
+            'Mar',
+            'Apr',
+            'Maj',
+            'Jun',
+            'Jul',
+            'Avg',
+            'Sep',
+            'Okt',
+            'Nov',
+            'Dec'
+        ],
+        years: []
+    };
+
+    componentDidMount() {
+        this.props.handleChangeTableData(this.state.date);
+
+        if (this.state.years.length === 0) {
+            let years = [];
+            for (let y = 2000; y <= getYear(new Date()); y++) {
+                years.push(y);
+            }
+            this.setState({ years: years });
+        }
+    }
+
+    handleOnDateClick = event => {
+        let clickedDay = event.target.textContent;
+        let currentDate = this.state.date;
+        this.setState(
+            {
+                date: addDays(currentDate, clickedDay - currentDate.getDate())
+            },
+            () => this.props.handleChangeTableData(this.state.date)
+        );
+    };
+
+    handleChangeDate = event => {
+        let option = event.target.textContent;
+        if (option === '-') {
+            let currentDate = this.state.date;
+            this.setState({ date: subMonths(currentDate, 1) }, () =>
+                this.props.handleChangeTableData(this.state.date)
+            );
+        } else if (option === '+') {
```

Comment:
Instead of many if-else statements, wouldn't it be easier to have a method for going forward 1 month bound to the "+" button, and a method for going back 1 month bound to the "-" button?